general_autoencoder.py

```diff
@@ -142,6 +143,12 @@ def decode(self, Z):
     def get_loss():
         raise NotImplementedError
 
+    def get_longitudinal_loss():
+        raise NotImplementedError
+
+    def get_Z1_from_Z0(self, Z0):
```

Comment:
we have a fast_forward_Z method -- how will this interact with it, and are we introducing redundancy?

Reply:
Hm. I tried to reconcile them, and realized that it's a bit harder than I thought. `get_Z1_from_Z0` is defining a new tensor, whereas our `fast_forward_Z` method takes in actual numbers and adjusts them. I'll add a comment to this effect; perhaps it's possible to treat them more coherently but not a huge deal now I think.